subprojects/core/src/main/java/org/gradle/internal/execution/TaskGraphBuildExecutionAction.java

```diff
@@ -39,28 +40,41 @@
  * A {@link BuildWorkExecutor} that does not execute any tasks, but prints the task graph instead.
  */
 public class TaskGraphBuildExecutionAction implements BuildWorkExecutor {
+    private final BuildWorkExecutor delegate;
     private final StyledTextOutputFactory textOutputFactory;
-
-    public TaskGraphBuildExecutionAction(StyledTextOutputFactory textOutputFactory) {
+    private final ConfigurationTimeBarrier configurationTimeBarrier;
+
+    public TaskGraphBuildExecutionAction(
+        BuildWorkExecutor delegate,
+        StyledTextOutputFactory textOutputFactory,
+        ConfigurationTimeBarrier configurationTimeBarrier
+    ) {
+        this.delegate = delegate;
         this.textOutputFactory = textOutputFactory;
+        this.configurationTimeBarrier = configurationTimeBarrier;
     }
 
     @Override
     public ExecutionResult<Void> execute(GradleInternal gradle, FinalizedExecutionPlan plan) {
+        if (configurationTimeBarrier.isAtConfigurationTime()) {
+            return delegate.execute(gradle, plan);
+        }
         StyledTextOutput output = textOutputFactory.create(TaskGraphBuildExecutionAction.class);
```

Comment:
💭  There is no need to create the output, unless we start rendering. It's good form to move this instantiation as close as possible to the use site